### PR TITLE
Np 2261 cristin migration

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -6,6 +6,7 @@ plugins {
     id 'base'
     id 'nebula.lint' version '16.22.0'
     id "maven"
+    id "io.freefair.lombok" version "5.3.0" apply false
```
Comment (axthosarouris) [inline build.gradle:9]: temporary. will remove later.

File: cristin-import/src/main/java/no/unit/nva/cristin/CristinMapper.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+import no.unit.nva.model.AdditionalIdentifier;
+import no.unit.nva.model.EntityDescription;
+import no.unit.nva.model.Publication;
+import no.unit.nva.model.PublicationDate;
+
+public class CristinMapper {
```
Comment (axthosarouris) [inline cristin-import/src/main/java/no/unit/nva/cristin/CristinMapper.java:15]: A template of a mapping the two versions of publication. The basic pattern will be 
```
streamOfCristinObjects.map(input->new CristinMapper(input)).map(mapper->mapper.generatePublication())
```

File: cristin-import/src/main/java/no/unit/nva/cristin/CristinMapper.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    public CristinMapper(CristinObject cristinObject) {
+        this.cristinObject = cristinObject;
+    }
+
+    public Publication generatePublication() {
+        return new Publication.Builder()
+                   .withAdditionalIdentifiers(Set.of(extractIdentifier()))
+                   .withEntityDescription(generateEntityDescription())
+                   .withCreatedDate(extractEntryCreationDate())
```
Comment (axthosarouris) [inline cristin-import/src/main/java/no/unit/nva/cristin/CristinMapper.java:27]: Some initial conversions. It is not certain if they will remain. I am mostly interested in the conversion pattern

File: cristin-import/src/main/java/no/unit/nva/cristin/CristinObject.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.List;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+@AllArgsConstructor(access = AccessLevel.PACKAGE)
+public class CristinObject {
```
Comment (axthosarouris) [inline cristin-import/src/main/java/no/unit/nva/cristin/CristinObject.java:14]: The object that will hold the information from Cristin. I think this is going to be a manual mapping between the json fields and the java fields

File: cristin-import/src/test/java/no/unit/nva/cristin/CristinMapperTest.java [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+public class CristinMapperTest {
+
+    public static final String SAMPLE_INPUT_01 = "input01";
+    private static final Integer NUMBER_OF_LINES_IN_RESOURCES_FILE = 100;
+
+    @Test
+    public void mapReturnsResourceWithCristinIdStoredInAdditionalIdentifiers() throws IOException {
```
Comment (axthosarouris) [inline cristin-import/src/test/java/no/unit/nva/cristin/CristinMapperTest.java:40]: Tests will describe the conversion of one field to another and cover the corner cases.

File: storage-model/src/main/java/no/unit/nva/publication/storage/model/Resource.java [hunk 6]
```diff
@@ -94,6 +100,7 @@ public static Resource fromPublication(Publication publication) {
                    .withEntityDescription(publication.getEntityDescription())
                    .withDoi(publication.getDoi())
                    .withHandle(publication.getHandle())
+                   .withAdditionalIdentifiers(publication.getAdditionalIdentifiers())
```
Comment (axthosarouris) [inline storage-model/src/main/java/no/unit/nva/publication/storage/model/Resource.java:103]: addiitonal identifiers added to Resource DAOs

**Review by brinxmat — APPROVED** [review]
